test/new-e2e/tests/gpu/gpu_test.go

```diff
@@ -0,0 +1,107 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package gpu
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/DataDog/test-infra-definitions/components/os"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/client"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client/agentclient"
+)
+
+var devMode = flag.Bool("devmode", false, "enable dev mode")
+
+type gpuSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+const defaultGpuCheckConfig = `
+init_config:
+  min_collection_interval: 5
+
+instances:
+  - {}
+`
+
+const defaultSysprobeConfig = `
+gpu_monitoring:
+  enabled: true
+`
+
+const vectorAddDockerImg = "nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda10.2"
+const gpuEnabledAMI = "ami-0f71e237bb2ba34be" // Ubuntu 22.04 with GPU drivers
+
+// TestGPUSuite runs tests for the VM interface to ensure its implementation is correct.
+func TestGPUSuite(t *testing.T) {
+	provisioner := awshost.Provisioner(
+		awshost.WithEC2InstanceOptions(
+			ec2.WithInstanceType("g4dn.xlarge"),
```

Comment:
I guess this is the smallest g4dn instance we can get?

Reply:
Yes, there's no smaller instance with GPUs.